### PR TITLE
feat: [CDS-70417]: Added labels support for V1 Pipeline Yaml

### PR DESCRIPTION
**Pull request by mohitg0795**
## Describe your changes
- Adding labels support in V1 pipeline yaml
- Integration of labels in execution flows
- Tags should be resolved only for V0 yaml and labels should be resolved only for V1 yaml

Test cases:
- Tested usual pipeline v0 execution with tags, working fine with all tags visible in execution history
- Tested V1 execution with labels, working fine
- Test runtime labels for V1 executions, working fine
- Test runtime tags for V0 executions, working fine

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49127)

### CODE REVIEW

**Comment by sahilhindwani-harness** [comment]
*[`884-pms-commons/src/test/java/io/harness/pms/utils/SimplifiedPipelineYamlHelperTest.java` line 36 at r3](https://reviewable.io/reviews/harness/harness-core/49127#-NXoVw8xA5X8Nnu15mjq:-NXoVw8y9ikq2O1nqMmB:bsy632n) ([raw file](https://github.com/harness/harness-core/blob/978941d4fc3f0796c42447c720b501ed896032bc/884-pms-commons/src/test/java/io/harness/pms/utils/SimplifiedPipelineYamlHelperTest.java#L36)):*
> ```Java
> 
> @OwnedBy(HarnessTeam.PIPELINE)
> public class SimplifiedPipelineYamlHelperTest extends CategoryTest {
> ```

nit: why rename?

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/labels/OrchestrationEndLabelsResolveHandler.java` line 33 at r3](https://reviewable.io/reviews/harness/harness-core/49127#-NXoW_TgEcX6BmDb7UNT:-NXoWOHc0KiUb8vFBRR_:bkk04wm) ([raw file](https://github.com/harness/harness-core/blob/978941d4fc3f0796c42447c720b501ed896032bc/pipeline-service/service/src/main/java/io/harness/pms/labels/OrchestrationEndLabelsResolveHandler.java#L33)):*
> ```Java
> @OwnedBy(HarnessTeam.PIPELINE)
> @Singleton
> public class OrchestrationEndLabelsResolveHandler implements OrchestrationEndObserver, AsyncInformObserver {
> ```

Can we merge this with OrchestrationTagsResolveHandler?
Otherwise 2 db calls will be there to fetch entity on end.

**Comment by mohitg0795** [comment]
*[`884-pms-commons/src/test/java/io/harness/pms/utils/SimplifiedPipelineYamlHelperTest.java` line 36 at r3](https://reviewable.io/reviews/harness/harness-core/49127#-NXoVw8xA5X8Nnu15mjq:-NXoaZjb1eMzSw6LDG9U:bkhj738) ([raw file](https://github.com/harness/harness-core/blob/978941d4fc3f0796c42447c720b501ed896032bc/884-pms-commons/src/test/java/io/harness/pms/utils/SimplifiedPipelineYamlHelperTest.java#L36)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

nit: why rename?
</blockquote></details>

Earlier it was named :PipelineYaml and if I use it anywhere or write helpers for it, it will become too confusing as its pretty generic name. Its analogous to our BasicPipeline entity that we had earlier. Thus renamed it for better context.
Please suggest if there's any better name that we can use here.

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/labels/OrchestrationEndLabelsResolveHandler.java` line 33 at r3](https://reviewable.io/reviews/harness/harness-core/49127#-NXoW_TgEcX6BmDb7UNT:-NXoap-x6zNOLTfzxEVC:b22f7yl) ([raw file](https://github.com/harness/harness-core/blob/978941d4fc3f0796c42447c720b501ed896032bc/pipeline-service/service/src/main/java/io/harness/pms/labels/OrchestrationEndLabelsResolveHandler.java#L33)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

Can we merge this with OrchestrationTagsResolveHandler?
Otherwise 2 db calls will be there to fetch entity on end. 
</blockquote></details>

Sure for now, these are simple operations so looks fine to combine. Will do so.

File: pipeline-service/service/src/main/java/io/harness/pms/helpers/LabelsHelper.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+@UtilityClass
+@Log4j
+public class LabelsHelper {
+  public List<NGLabel> getLabels(String yaml, String version) {
+    if (PipelineVersion.V1.equals(version)) {
+      List<NGLabel> ngLabelList = new ArrayList<>();
+      try {
+        SimplifiedPipelineYaml simplifiedPipelineYaml = SimplifiedPipelineYamlHelper.getSimplifiedPipelineYaml(yaml);
+        Map<String, String> labels = simplifiedPipelineYaml.getLabels();
```
Comment (ankittiwari-harness) [inline pipeline-service/service/src/main/java/io/harness/pms/helpers/LabelsHelper.java:28]: do we require null check ?

**Review by rohitk-h — APPROVED** [review]
> Reviewed 2 of 9 files at r1, 3 of 7 files at r2, 9 of 9 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49127)* status: 18 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 2 of 9 files at r1, 3 of 7 files at r2, 8 of 9 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49127)* status: 1 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/helpers/LabelsHelper.java` line 22 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXscu763gwSVP6yrBVj:-NXscu763gwSVP6yrBVk:b6zaofh) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/main/java/io/harness/pms/helpers/LabelsHelper.java#L22)):*
> ```Java
> @UtilityClass
> @Log4j
> public class LabelsHelper {
> ```

team annotation
___
*[`pipeline-service/service/src/test/java/io/harness/pms/helpers/LabelsHelperTest.java` line 28 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXsdMzU0ves3bR_3WKu:-NXsdMzU0ves3bR_3WKv:b6zaofh) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/test/java/io/harness/pms/helpers/LabelsHelperTest.java#L28)):*
> ```Java
> import org.junit.experimental.categories.Category;
> 
> public class LabelsHelperTest extends CategoryTest {
> ```

team annotation
___
*[`953-yaml-commons/src/main/java/io/harness/yaml/core/NGLabel.java` line 15 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXscoiJ5m1-3KUyOxUv:-NXscoiJ5m1-3KUyOxUw:b6zaofh) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/953-yaml-commons/src/main/java/io/harness/yaml/core/NGLabel.java#L15)):*
> ```Java
> @Data
> @Builder
> public class NGLabel {
> ```

team annotation
___
*[`pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java` line 17 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXsd-p74mS3zcPF-fyo:-NXsd-p74mS3zcPF-fyp:b6zaofh) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java#L17)):*
> ```Java
> 
> @UtilityClass
> public class SimplifiedPipelineYamlHelper {
> ```

team annotation
___
*[`pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java` line 17 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXsd4lq-Q1a9N8_fmZJ:-NXsd4lq-Q1a9N8_fmZK:bhznvqn) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java#L17)):*
> ```Java
> 
> @UtilityClass
> public class SimplifiedPipelineYamlHelper {
> ```

why separate helper is required? this YamlUTils.read exists call everywhere and we don't have helper per entity just for this as of now
___
*[`pipeline-service/service/src/test/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelperTest.java` line 24 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXsdRaDBDuvN27NLfGD:-NXsdRaDBDuvN27NLfGE:b6zaofh) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/test/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelperTest.java#L24)):*
> ```Java
> import org.junit.experimental.categories.Category;
> 
> public class SimplifiedPipelineYamlHelperTest extends CategoryTest {
> ```

team annotation

**Comment by mohitg0795** [comment]
*[`953-yaml-commons/src/main/java/io/harness/yaml/core/NGLabel.java` line 15 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXscoiJ5m1-3KUyOxUv:-NXsezzi4ISEQdyHZcQ9:bv3w1pl) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/953-yaml-commons/src/main/java/io/harness/yaml/core/NGLabel.java#L15)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

team annotation
</blockquote></details>

Do you have idea around use-case for team labels? I don't think it matters.

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java` line 17 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXsd4lq-Q1a9N8_fmZJ:-NXsfARd1ctsp0MM7_Z4:b-qwl2t) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java#L17)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

why separate helper is required? this YamlUTils.read exists call everywhere and we don't have helper per entity just for this as of now
</blockquote></details>

Currently, we have to write YamlUtils.read(yaml, classname.class). So, to avoid all this hassle, created a basic helper which works on hiding all these things and simply providers a wrapper implementation.

File: pipeline-service/service/src/main/java/io/harness/pms/helpers/LabelsHelper.java (hunk 1 above)
Comment (mohitg0795) [reply pipeline-service/service/src/main/java/io/harness/pms/helpers/LabelsHelper.java:28]: done

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 9 files at r3, 7 of 7 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49127)* status: Some of the checks are too old. 8 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/labels/OrchestrationEndLabelsResolveHandler.java` line 29 at r6](https://reviewable.io/reviews/harness/harness-core/49127#-NYIYB-pD7A12fjenFrf:-NYIYB-pD7A12fjenFrg:b-29uwfb) ([raw file](https://github.com/harness/harness-core/blob/e9446083c8d759fbad72cc2c3b3db26d1ad19d9e/pipeline-service/service/src/main/java/io/harness/pms/pipeline/labels/OrchestrationEndLabelsResolveHandler.java#L29)):*
> ```Java
> 
> @Singleton
> public class OrchestrationEndLabelsResolveHandler implements OrchestrationEndObserver, AsyncInformObserver {
> ```

Didnt we decide to move this code to OrchestrationEndTagsResolveHandler isn't?
___
*[`pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java` line 17 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXsd4lq-Q1a9N8_fmZJ:-NYIXvp00Ab7UgrW2mBE:bduog12) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java#L17)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Currently, we have to write YamlUtils.read(yaml, classname.class). So, to avoid all this hassle, created a basic helper which works on hiding all these things and simply providers a wrapper implementation.
</blockquote></details>

done, i dont think these are unnecessary wrappers, devs should be aware as its most critical about it

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49127)* status: Some of the checks are too old. 8 file(s) to review. 1 change requests. 8. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/pipeline/labels/OrchestrationEndLabelsResolveHandler.java` line 29 at r6](https://reviewable.io/reviews/harness/harness-core/49127#-NYIYB-pD7A12fjenFrf:-NYIZ9Y795JJ3xDoN7zV:bjse0ez) ([raw file](https://github.com/harness/harness-core/blob/e9446083c8d759fbad72cc2c3b3db26d1ad19d9e/pipeline-service/service/src/main/java/io/harness/pms/pipeline/labels/OrchestrationEndLabelsResolveHandler.java#L29)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

Didnt we decide to move this code to OrchestrationEndTagsResolveHandler isn't?
</blockquote></details>

Earlier our idea was that tags and labels can be fetched, resolved and saved in same query, thus it sounded correct. But actually, we don't do any operation on labels for V1 and we don't resolve any tags for V0 version of yaml, thus kept them separate along with their specific version checks as well.
___
*[`pipeline-service/service/src/test/java/io/harness/pms/helpers/LabelsHelperTest.java` line 28 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXsdMzU0ves3bR_3WKu:-NYIZUpVDqJvv3g4eSe3:bnj7hl0) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/test/java/io/harness/pms/helpers/LabelsHelperTest.java#L28)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

team annotation
</blockquote></details>

same as above
___
*[`pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java` line 17 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXsd-p74mS3zcPF-fyo:-NYIYl8UDR8vwHoD-Wtg:b-euirvw) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java#L17)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

team annotation
</blockquote></details>

I think its not required and used anywhere. We can check and add it if required.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java` line 17 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXsd4lq-Q1a9N8_fmZJ:-NYIYxoG6ErYNeB467kg:bhwqlqg) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/main/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelper.java#L17)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

done, i dont think these are unnecessary wrappers, devs should be aware as its most critical about it
</blockquote></details>

Sorry, may be I'm missing something here but I also feel that these are good wrapper implementations to have and simpler to use, thus added it here. 
___
*[`pipeline-service/service/src/test/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelperTest.java` line 24 at r4](https://reviewable.io/reviews/harness/harness-core/49127#-NXsdRaDBDuvN27NLfGD:-NYIZVhJ0g91bDOdn2FA:bnj7hl0) ([raw file](https://github.com/harness/harness-core/blob/4d693d1d474a4c16aadc3898e2789830d64e3c93/pipeline-service/service/src/test/java/io/harness/pms/helpers/SimplifiedPipelineYamlHelperTest.java#L24)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

team annotation
</blockquote></details>

same as above

**Review by rohitk-h — APPROVED** [review]
> Reviewed 7 of 7 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49127)* status: Some of the checks are too old. 3. unresolved discussions. 

___
*[`pipeline-service/service/src/test/java/io/harness/pms/labels/OrchestrationEndLabelsResolveHandlerTest.java` line 95 at r6](https://reviewable.io/reviews/harness/harness-core/49127#-NYKJtHND-BEKTLDM-mK:-NYKJtHND-BEKTLDM-mL:b-x9jq51) ([raw file](https://github.com/harness/harness-core/blob/5019e128cd7781dcf870028cab637c6d430425e2/pipeline-service/service/src/test/java/io/harness/pms/labels/OrchestrationEndLabelsResolveHandlerTest.java#L95)):*
> ```Java
>     PipelineExecutionSummaryEntity dummyEntity = PipelineExecutionSummaryEntity.builder()
>                                                      .label(NGLabel.builder().key("key").value("val").build())
>                                                      .pipelineVersion(PipelineVersion.V0)
> ```

nit: should we add unit test for pipeline with `.pipelineVersion` not set in the entity, as this is a new field?

**Review by ankittiwari-harness — APPROVED** [review]

File: pipeline-service/service/src/main/java/io/harness/pms/helpers/LabelsHelper.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.List;
+import java.util.Map;
+import lombok.experimental.UtilityClass;
+import lombok.extern.log4j.Log4j;
+
+@UtilityClass
+@Log4j
+public class LabelsHelper {
+  public List<NGLabel> getLabels(String yaml, String version) {
+    if (PipelineVersion.V1.equals(version)) {
```
[the remark this answers is not in the document]
Comment (mohitg0795) [reply pipeline-service/service/src/main/java/io/harness/pms/helpers/LabelsHelper.java:24]: image

We don't do this for V0 flow, thus no else block

**Comment by mohitg0795** [comment]
trigger uts

**Comment by mohitg0795** [comment]
trigger review

**Comment by mohitg0795** [comment]
trigger smartchecks

**Pull request merged**